src/wp-includes/formatting.php

```diff
@@ -5852,31 +5852,23 @@ function wp_spaces_regexp() {
  * @since 4.2.0
  */
 function print_emoji_styles() {
-	static $printed = false;
-
-	if ( $printed ) {
-		return;
-	}
-
-	$printed = true;
-
-	$type_attr = current_theme_supports( 'html5', 'style' ) ? '' : ' type="text/css"';
-	?>
-<style<?php echo $type_attr; ?>>
-img.wp-smiley,
-img.emoji {
-	display: inline !important;
-	border: none !important;
-	box-shadow: none !important;
-	height: 1em !important;
-	width: 1em !important;
-	margin: 0 0.07em !important;
-	vertical-align: -0.1em !important;
-	background: none !important;
-	padding: 0 !important;
-}
-</style>
-	<?php
+	$emoji_styles = '
+	img.wp-smiley,
+	img.emoji {
+		display: inline !important;
+		border: none !important;
+		box-shadow: none !important;
+		height: 1em !important;
+		width: 1em !important;
+		margin: 0 0.07em !important;
+		vertical-align: -0.1em !important;
+		background: none !important;
+		padding: 0 !important;
+	}';
+	$handle       = 'wp-emoji-styles';
+	wp_register_style( $handle, false );
+	wp_add_inline_style( $handle, $emoji_styles );
+	wp_enqueue_style( $handle );
```

Comment:
This change is _probably_ safe, although it could break a plugin that tries to do something like:

```php
ob_start();
print_emoji_styles();
$css = ob_get_clean();
```

Reply:
This is the case for many of the changes in the PR. This will need to be called out in the dev note.

Reply:
Similar to my above comment: I think we should leave the old function as is and deprecate it. Deprecating is fine, making a breaking change is not, particularly when there is a straightforward way not to break things. Explaining a breaking change in a dev note should still be a last resort, better not to introduce one. Additionally, the current solution doesn't 100% appropriately implement the solution since it enqueues the styles in a function that is hooked into `wp_print_styles`. Yes, it technically works, but as the name states enqueuing should preferably happen in `wp_enqueue_scripts`. So I think this solution tries to be somewhere in between backward compatibility and properly enqueuing but doesn't properly achieve either.

A simple solution is to deprecate `print_emoji_styles()` and instead introduce `enqueue_emoji_styles()`.